devices/google-marlin/default.nix

```diff
@@ -0,0 +1,51 @@
+{ config, lib, pkgs, ... }:
+
+{
+  mobile.device.name = "google-marlin";
+  mobile.device.info = rec {
+    name = "Google Pixel XL";
+    # TODO : make kernel part of options.
+    kernel = pkgs.callPackage ./kernel { kernelPatches = pkgs.defaultKernelPatches; };
+
+    format_version = "0";
+    manufacturer = "Google";
+    codename = "google-marlin";
+    date = "";
+    dtb = "";
+    modules_initfs = "";
+    arch = "aarch64";
+    keyboard = "false";
+    external_storage = "true";
+    screen_width = "1440";
+    screen_height = "2880";
+    dev_touchscreen = "";
+    dev_touchscreen_calibration = "";
+    dev_keyboard = "";
+    flash_method = "fastboot";
+
+    kernel_cmdline = "console=ttyHSL0,115200,n8 androidboot.console=ttyHSL0 androidboot.hardware=marlin user_debug=31 ehci-hcd.park=3 lpm_levels.sleep_disabled=1 cma=32M@0-0xffffffff loop.max_part=7 buildvariant=eng";
+
+    generate_bootimg = true;
+    bootimg_qcdt = false;
+    flash_offset_base = "0x80000000";
+    flash_offset_kernel = "0x00008000";
+    flash_offset_ramdisk = "0x01000000";
+    flash_offset_second = "0x00f00000";
+    flash_offset_tags = "0x00000100";
+    flash_pagesize = "4096";
+  };
+
+  mobile.hardware = {
+    soc = "qualcomm-msm8996";
+    ram = 1024 * 4;
+    screen = {
+      width = 1440; height = 2880;
+    };
+  };
+
+  mobile.system.type = "android";
+
+  # FIXME: properly use the partlabel instead.
+  # This is a huge hack and should be dropped ASAP.
+  #fileSystems."/".device = lib.mkForce "/dev/sda13";
```

Comment:
These lines can be removed :). For `google-taimen` it was likely an issue with udev that was resolved since the PR was opened, as it just works with `google-walleye` and other devices.